devp2p/kademlia.py

```diff
@@ -480,17 +484,17 @@ def _expected_pongs():
             rid = random.randint(bucket.start, bucket.end)
             self.find_node(rid)
 
-        # check and removed timedout find requests
-        for nodeid, timeout in self._find_requests.items():
+        # check and removed timed out find requests
+        for nodeid, timeout in tuple(self._find_requests.items()):
```

Comment:
Another nitpick.  

I dislike that this dictionary is modified in place.  Mutability leads to terrible things and whenever possible I always try to avoid it.

It could be replaced with a comprehension to replace it with an entirely new dictionary which would remove part the mutation aspect.

```python
self._find_requests = {
    nodeid: timeout
    for nodeid, timeout in self._find_requests.items()
    if time.time() <= timeout
}
```